protocol/test/forge/v2/templeLineOfCredit/TlcBorrow.t.sol

```diff
@@ -336,50 +125,108 @@ contract TempleLineOfCreditTestBorrow is TlcBaseTest {
 
         uint256 borrowAmount = 50_000e18;
         vm.startPrank(alice);
-        tlc.requestBorrow(borrowAmount);
-        tlc.requestBorrow(borrowAmount);
-        vm.warp(block.timestamp + BORROW_REQUEST_MIN_SECS);
 
         vm.expectEmit(address(tlc));
         emit Borrow(alice, bob, borrowAmount);
-        tlc.borrow(bob);
+        tlc.borrow(borrowAmount, bob);
         
         assertEq(daiToken.balanceOf(alice), 0);
         assertEq(daiToken.balanceOf(bob), borrowAmount);
     }
 
-    function test_borrow_noWait() public {
+    function test_pause_borrow() public {
         uint256 collateralAmount = 100_000e18;
         addCollateral(alice, collateralAmount);
 
-        vm.prank(executor);
-        tlc.setBorrowRequestConfig(0, 0);
+        vm.startPrank(executor);
+        tlc.setBorrowPaused(true);
 
         uint256 borrowAmount = 50_000e18;
-        vm.startPrank(alice);
-        tlc.requestBorrow(borrowAmount);
+        changePrank(alice);
 
-        vm.expectEmit(address(tlc));
-        emit Borrow(alice, alice, borrowAmount);
-        tlc.borrow(alice);
+        vm.expectRevert(abi.encodeWithSelector(Paused.selector));
+        tlc.borrow(borrowAmount, alice);
+
+        // Unpause
+        changePrank(executor);
+        tlc.setBorrowPaused(false);
+
+        changePrank(alice);
+        tlc.borrow(borrowAmount, alice);
     }
 
-    function test_pause_borrow() public {
+    function test_borrow_rescueMode() public {
         uint256 collateralAmount = 100_000e18;
+        uint256 borrowAmount = 50_000e18;
         addCollateral(alice, collateralAmount);
 
-        vm.prank(executor);
-        tlc.setBorrowRequestConfig(1, 0);
+        {
+            vm.startPrank(rescuer);
+            tlc.setRescueMode(true);
 
-        uint256 borrowAmount = 50_000e18;
-        vm.startPrank(alice);
-        tlc.requestBorrow(borrowAmount);
+            changePrank(alice);
+            vm.expectRevert(abi.encodeWithSelector(CommonEventsAndErrors.InvalidAccess.selector));
```

Comment:
nit: could we give the custom error in `notInRescueMode()` a more specific name?

Reply:
Currently it's consistent within the `TempleElevatedAccess` to throw an InvalidAccess:

https://github.com/TempleDAO/temple/blob/temple-v2-review-feedback/protocol/contracts/v2/access/TempleElevatedAccess.sol#L130-L143

I think it would make it messy if we add a different custom error as isElevatedAccess would have to be tweaked